src/main/java/org/opentripplanner/middleware/utils/ApiGatewayUtils.java

```diff
@@ -86,23 +93,28 @@ public static ApiKey createApiKey(ApiUser user, String usagePlanId) throws Creat
                 .withTags(Collections.singletonMap("userId", user.id))
                 .withEnabled(true)
                 .withSdkRequestTimeout(SDK_REQUEST_TIMEOUT);
-            CreateApiKeyResult apiKeyResult = gateway.createApiKey(apiKeyRequest);
+            apiKeyResult = gateway.createApiKey(apiKeyRequest);
+            if (apiKeyResult != null) keyId = apiKeyResult.getId();
+            LOG.info("Created new API key {}", keyId);
+        } catch (Exception e) {
+            CreateApiKeyException createApiKeyException = new CreateApiKeyException(user.id, usagePlanId, e);
 
-            // add API key to usage plan
-            CreateUsagePlanKeyRequest usagePlanKeyRequest = new CreateUsagePlanKeyRequest();
-            usagePlanKeyRequest
+            BugsnagReporter.reportErrorToBugsnag("Error creating API key", keyId, createApiKeyException);
+            throw createApiKeyException;
+        }
+        try {
```

Comment:
I don't quite understand why there are 2 try/catch blocks here. It seems like there should be just one try/catch and the `catch (Exception e)` block should be retained in almost the same position where it was, only this time it'd be included right after the `catch (ConflictException e)` block. This way it seems that any Exception generated from the `gateway.createUsagePlanKey(usagePlanKeyRequest)` call that isn't a `ConflictException` will be properly caught.

Reply:
I don't know if ConflictExceptions can be thrown by `AmazonApiGateway#createApiKey`. There may be ConflictExceptions that we want to cause a `CreateApiKeyException` rather than just continue with CreateUsagePlanKeyRequest. I'd rather err on the side of caution and not try to bundle these calls up into the same block.

Reply:
Ok, that reasoning sounds good enough to keep the 2 try/catch blocks. However, I think 2 improvements should be made:

1. Add some comments about this decision
2. Add another `catch(Exception e)` block to the 2nd try/catch block to make sure that any other funky AWS client error is properly caught and reported.

Reply:
See nested solution in dc6630e8a2470cd01f38a49a62c3d86c554dfde5